### ISSUE TITLE
[menu-bar] Access to the overflow menu

### ISSUE THREAD

**Issue by anezthes** [vaadin/flow-components#2988]

**Comment by rolfsmeds** [comment]
So the use case you're solving for here is when you want some menu items to _always_ be in a menu, and for that same menu to function as the overflow menu?

In other words, the solution could be either to add items directly to the overflow menu (which also forces it to be always visible), _or_ to designate any top-level menu item to function as the overflow menu.

**Comment by jouni** [comment]
I interpret this requirement as "max number of items visible". For example, if I have items `[ one | two | three | four | five ]`, and define that max 2 should be visible at any time, I would get 

```
[ one | two | ··· ]
              [ three ]
              [ four  ]
              [ five  ]
```

**Comment by web-padawan** [comment]
We have `maxItemsVisible` API in `vaadin-avatar-group` so it would be natural to introduce it to `vaadin-menu-bar`.

**Comment by rolfsmeds** [comment]
Not so sure about that – typically you want _certain items_ to be in the bar, and _certain others_ in the dropdown, not a certain _number_ of items to be visible. E.g. you want the primary actions in the menubar to be visible (if there's room) and the secondary items in a dropdown (and then use that same dropdown for the overflow).

So, while you _can_, in most cases, do that with `maxItemsVisible`, you're forced to play a counting game instead of just telling the component exactly how you want your items, and in many cases the menubar is dynamically populated (e.g. based on a selection in a Grid) so you don't even know how many items there are (or you have to update that property each time one is added).

**Comment by jouni** [comment]
Yeah, a number of items makes less sense in Menu Bar than it does in Avatar Group. Perhaps you should be able to mark certain items to always be collapsed instead?

**Comment by web-padawan** [comment]
> Perhaps you should be able to mark certain items to always be collapsed instead?

Yes, sounds like a proper way to do that. We would need some kind of flag to distinguish such items from the normal items that can be shown when there is enough space. It could be a bit tricky if user wants to set this dynamically.

**Comment by rolfsmeds** [comment]
That, OR, designate a regular top-level menu to be _the_ overflow menu. Although I suspect that might more challenging to implement?
